tools/pymanifold.cpp

```diff
@@ -0,0 +1,169 @@
+
+namespace py = pybind11;
+
+using namespace manifold;
+
+void exportManifold(Manifold &m, std::string name) {
+  Mesh out = m.GetMesh();
+  manifold::ExportOptions options;
+  options.faceted = true;
+  options.mat.roughness = 0.2;
+  options.mat.metalness = 0.0;
+  manifold::ExportMesh(name, out, options);
+};
+
+typedef std::tuple<float, float> Float2;
+typedef std::tuple<float, float, float> Float3;
+
+struct PolygonsWrapper {
+  std::unique_ptr<Polygons> polygons;
+};
+
+PYBIND11_MODULE(pymanifold, m) {
+  m.doc() = "Python binding for the manifold library. Please check the C++ documentation for APIs.\n"
+      "This binding will perform copying to make the API more familiar to OpenSCAD users.";
+  py::class_<Manifold>(m, "Manifold")
+      .def(py::init<>())
+      .def(py::init([](std::vector<Manifold> &manifolds) {
+        Manifold result;
+        for (Manifold &manifold : manifolds)
+          result += manifold;
+        return result;
+      }), "Construct manifold as the union of a set of manifolds.")
+      .def(py::self + py::self)
+      .def(py::self - py::self)
+      .def("__and__", [](Manifold &a, Manifold &b) { return a ^ b; })
+      .def("transform",
+           [](Manifold &self, py::array_t<float> &mat) {
+             auto mat_view = mat.unchecked<2>();
+             if (mat_view.shape(0) != 3 || mat_view.shape(1) != 4)
+               throw std::runtime_error("Invalid matrix shape");
+             glm::mat4x3 mat_glm;
+             for (int i = 0; i < 3; i++) {
+               for (int j = 0; j < 4; j++) {
+                 mat_glm[j][i] = mat_view(i, j);
+               }
+             }
+             return self.Transform(mat_glm);
+           })
+      .def(
+          "translate",
+          [](Manifold self, float x, float y, float z = 0.0f) {
+            return self.Translate(glm::vec3(x, y, z));
+          },
+          py::arg("x"), py::arg("y"), py::arg("z") = 0.0f)
+      .def("scale", static_cast<Manifold (*)(Manifold, float)>(
+                        [](Manifold self, float scale) {
+                          return self.Scale(glm::vec3(scale));
+                        }), py::arg("scale"))
+      .def("scale", static_cast<Manifold (*)(Manifold, py::array_t<float> &)>(
+                        [](Manifold self, py::array_t<float> &scale) {
+                          auto scale_view = scale.unchecked<1>();
+                          if (scale_view.shape(0) != 3)
+                            throw std::runtime_error("Invalid vector shape");
+                          glm::vec3 v(scale_view(0), scale_view(1),
+                                      scale_view(2));
+                          return self.Scale(v);
+                        }), py::arg("v"))
+      .def("rotate",
+           [](Manifold self, float xDegrees, float yDegrees, float zDegrees) {
+             return self.Rotate(xDegrees, yDegrees, zDegrees);
+           }, py::arg("xDegrees"), py::arg("yDegrees"), py::arg("zDegrees"))
```

Comment:
Should these have `= 0` so that they don't all need to be specified?

Reply:
yes, will change